src/omv/ports/mimxrt/omv_spi.c

```diff
@@ -78,43 +89,81 @@ static void spi_master_callback(LPSPI_Type *base, void *handle, status_t status,
     } else {
         spi->xfer_flags |= OMV_SPI_XFER_FAILED;
         spi->xfer_error = status;
-        if (spi->xfer_flags & OMV_SPI_XFER_DMA) {
-            LPSPI_MasterTransferAbortEDMA(spi->inst, &spi->descr_master_edma);
-        } else {
-            LPSPI_MasterTransferAbort(spi->inst, &spi->descr_master);
+        omv_spi_transfer_abort(spi);
+    }
+
+    void *buf = spi->xfer_descr.rxData;
+
+    if (buf == NULL) {
+        buf = spi->xfer_descr.txData;
+    }
+
+    // The IMXRT doesn't support half complete transfer interrupts (in the lpspi driver) like the STM32
+    // does. So, mimick support for them by toggling the half flag.
+    uint32_t flags = (OMV_SPI_XFER_DMA | OMV_SPI_XFER_COMPLETE);
+    if ((spi->dma_flags & OMV_SPI_DMA_DOUBLE) && ((spi->xfer_flags & flags) == flags)) {
+        int32_t offset = (spi->xfer_flags & OMV_SPI_XFER_HALF) ? -spi->xfer_descr.dataSize : spi->xfer_descr.dataSize;
+        if (spi->xfer_descr.txData) {
+            spi->xfer_descr.txData += offset;
```

Comment:
```suggestion
```
Since this is no longer a valid mode, I don't think we need this.

Reply:
Done